internal/blockchain/ethereum/quorum/geth_provider.go

```diff
@@ -0,0 +1,409 @@
+		address := account.(*ethereum.Account).Address
+		l.Info(fmt.Sprintf("unlocking account %s", address))
+		// Check which member the account belongs to
+		var memberIndex int
+		for _, member := range p.stack.Members {
+			if member.Account.(*ethereum.Account).Address == address {
+				memberIndex = *member.Index
+			}
+		}
+		if err := p.unlockAccount(address, keyPassword, memberIndex); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (p *GethProvider) unlockAccount(address, password string, memberIndex int) error {
+	l := log.LoggerFromContext(p.ctx)
+	verbose := log.VerbosityFromContext(p.ctx)
+	// exposed blockchain port is the default for node 0, we need to add the port multiplier to get the right rpc for the correct node
+	gethClient := NewGethClient(fmt.Sprintf("http://127.0.0.1:%v", p.stack.ExposedBlockchainPort+(memberIndex*exposedBlockchainPortMultiplier)))
+	retries := 10
+	for {
+		if err := gethClient.UnlockAccount(address, password); err != nil {
+			if verbose {
+				l.Debug(err.Error())
+			}
+			if retries == 0 {
+				return fmt.Errorf("unable to unlock account %s", address)
+			}
+			time.Sleep(time.Second * 1)
+			retries--
+		} else {
+			break
+		}
+	}
+	return nil
+}
+
+func (p *GethProvider) DeployFireFlyContract() (*types.ContractDeploymentResult, error) {
+	contract, err := ethereum.ReadFireFlyContract(p.ctx, p.stack)
+	if err != nil {
+		return nil, err
+	}
+	return p.connector.DeployContract(contract, "FireFly", p.stack.Members[0], nil)
+}
+
+func (p *GethProvider) GetDockerServiceDefinitions() []*docker.ServiceDefinition {
+	memberCount := len(p.stack.Members)
+	serviceDefinitionsCount := memberCount
+	if p.stack.TesseraEnabled {
+		serviceDefinitionsCount *= 2
+	}
+	serviceDefinitions := make([]*docker.ServiceDefinition, serviceDefinitionsCount)
+	connectorDependents := map[string]string{}
+	for i := 0; i < memberCount; i++ {
+		var dependsOn map[string]map[string]string
+		if p.stack.TesseraEnabled {
+			dependsOn = map[string]map[string]string{fmt.Sprintf("tessera_%d", i): {"condition": "service_started"}}
+		}
+		serviceDefinitions[i] = &docker.ServiceDefinition{
+			ServiceName: fmt.Sprintf("geth_%d", i),
+			Service: &docker.Service{
+				Image:         gethImage,
+				ContainerName: fmt.Sprintf("%s_geth_%d", p.stack.Name, i),
+				Volumes:       []string{fmt.Sprintf("geth_%d:/data", i)},
+				Logging:       docker.StandardLogOptions,
+				Ports:         []string{fmt.Sprintf("%d:8545", p.stack.ExposedBlockchainPort+(i*exposedBlockchainPortMultiplier))}, // defaults 5100, 5110, 5120, 5130
+				Environment:   p.stack.EnvironmentVars,
+				EntryPoint:    []string{"/bin/sh", "-c", "/data/docker-entrypoint.sh"},
+				DependsOn:     dependsOn,
+			},
+			VolumeNames: []string{fmt.Sprintf("geth_%d", i)},
+		}
+		connectorDependents[fmt.Sprintf("geth_%d", i)] = "service_started"
+
+		if p.stack.TesseraEnabled {
+			serviceDefinitions[i+memberCount] = &docker.ServiceDefinition{
```

Comment:
nit move depends on lower
```suggestion
		serviceDefinitions[i] = &docker.ServiceDefinition{
			ServiceName: fmt.Sprintf("geth_%d", i),
			Service: &docker.Service{
				Image:         gethImage,
				ContainerName: fmt.Sprintf("%s_geth_%d", p.stack.Name, i),
				Volumes:       []string{fmt.Sprintf("geth_%d:/data", i)},
				Logging:       docker.StandardLogOptions,
				Ports:         []string{fmt.Sprintf("%d:8545", p.stack.ExposedBlockchainPort+(i*exposedBlockchainPortMultiplier))}, // defaults 5100, 5110, 5120, 5130
				Environment:   p.stack.EnvironmentVars,
				EntryPoint:    []string{"/bin/sh", "-c", "/data/docker-entrypoint.sh"},
				DependsOn:     dependsOn,
			},
			VolumeNames: []string{fmt.Sprintf("geth_%d", i)},
		}
		connectorDependents[fmt.Sprintf("geth_%d", i)] = "service_started"

		if p.stack.TesseraEnabled {
		        dependsOn = map[string]map[string]string{fmt.Sprintf("tessera_%d", i): {"condition": "service_started"}}
			serviceDefinitions[i+memberCount] = &docker.ServiceDefinition{
```

Reply:
quorum containers have a dependencies to tessera, shifted quorum docker service lower to avoid duplicating tesseraEnabled checks